components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1977,7 +1977,18 @@ do_default_build_config() {
   _do_copy_templates "config"
   _do_copy_templates "config_install"
   if [[ -d "$PLAN_CONTEXT/hooks" ]]; then
-    cp -r "$PLAN_CONTEXT/hooks" "$pkg_prefix"
+    mkdir -p "$pkg_prefix"/hooks
+    for file in "$PLAN_CONTEXT"/hooks/*
+    do
+      # If there are multiple files for the same hook but different extensions
```

Comment:
Rather than guess, I wonder if we want to bail if a user specifies more than one hook?  I don't have a good idea on how to do that on the Linux side though since your hook language can be arbitrary (including .ps1).

Reply:
I like the idea of erroring when encountering multiple hooks. If any files in the hooks directory have the same filename sans the extension it would be an error.

This would also prevent a minor regression. If we prefer an extensioned hook (eg `ps1` on windows, and `sh` on linux) if a user had a `<hook>` and `<hook>.<sh|ps1>` files this PR would now use the extensioned version whereas the current behavior would use the non-extensioned version.

Reply:
ok this should be good to go now @smacfarlane and @davidMcneil